Addition of MoleculeVisualizer Class

talagayev opened the pull request:
Changes Made:

Following files were created:
- `mdonatello.py` created with following classes:
  - `MoleculeVisualizer` A class for small molecule 2D visualization in jupyter notebook consisting of:

    -  `__init__` , which initializes the MoleculeVisualizer with an AtomGroup and visualization options.
    - `initialize_widgets`, which initializes the interactive widgets for molecule visualization.
    - `initialize_output`, which initializes the output display and arranges the widgets in the interface.
    - `link_widget_callbacks`, which links the interactive widgets to the update_display method to reflect changes.
    - `update_display`, which updates the molecule display based on the current selections.
    - `save_selected_molecule`, which saves the currently selected molecule as a PNG file.

- `drawer.py` created with following classes:
  - `PharmacophoreHighlighter` A class responsible for highlighting of the pharmacophore features in a molecule consisting of:
    
    - `__init__` , which initializes the PharmacophoreHighlighter with a molecule, pharmacophore checkboxes, and an RDKit feature factory.
    - `determine_highlights`, which determines the atom and bond indices to highlight them based on the selected pharmacophore features.

  - `FunctionalGroupHighlighter` A class responsible for highlighting of the functional groups in a molecule consisting of:
    - `__init__` , which initializes the FunctionalGroupHighlighter with a molecule and functional group checkboxes.
    - `determine_highlights` , which determines the atom and bond indices to highlight based on the selected functional groups.

  - `RotatableBondsHighlighter` A class responsible for highlighting of rotatable bonds in a molecule consisting of:
    - `__init__` , which initializes the RotatableBondsHighlighter with a molecule and a checkbox for rotatable bonds.
    - `determine_highlights` , which determines the atom and bond indices to highlight them based on the presence of rotatable bonds.

  - `PartialChargeHighlighter` A class responsible for highlighting of partial charges in a molecule consisting of:
    - `__init__` , which initializes the PartialChargeHighlighter with a molecule and a checkbox for partial charges.
    - `determine_highlights` , which determines the atom and bond indices to highlight them based on the partial charges.

  - `StereocenterHighlighter` A class responsible for highlighting of stereocenters in a molecule consisting of:
    - `__init__` , which initializes the StereocenterHighlighter with a molecule and a checkbox for stereocenters.
    - `determine_highlights` , which determines the atom and bond indices to highlight based on their stereocenter configuration.

  - `MurckoScaffoldHighlighter` A class responsible for highlighting of the Murcko scaffold in a molecule consisting of:
    - `__init__` , which initializes the MurckoScaffoldHighlighter with a molecule and a checkbox for Murcko scaffolds.
    - `determine_highlights` , which determines the atom and bond indices to highlight them based on the presence of Murcko scaffolds.

  - `MoleculeDrawer` A class for drawing a molecule with various highlighted features consisting of:
    - `__init__` , which initializes the MoleculeDrawer with a molecule and various highlighting checkboxes and features.
    - `draw_molecule` , which draws the molecule with the highlighted features.

- `mapper.py` created with following classes:
  - `FunctionalGroupHandler` A class that is responsible for handling functional groups in molecules consisting of:
    - `calculate_functional_groups` , which identifies, calculates and counts the functional groups present in the given RDKit molecule.
    - `get_color_for_functional_group` , which determines the color associated with the given functional group based on its chemical structure.

  - `PharmacophoreColorMapper` A class for mapping pharmacophore features according to their respective colors consisting of:
    - `get_color_for_pharmacophore` , which determines the color associated with the given pharmacophore feature family.

- `properties.py` created with following classes:
  - `Property` A base class for representing a property of a molecule consisting of:
    - `__init__` , which initializes the Property class with a molecule.
    - `property_value` , which is abstract method to be implemented by subclasses to calculate the property value.
    - `__repr__` , which returns the HTML-formatted string representation of the property name and value..

  - `MolecularWeight` A class for calculating the molecular weight of a molecule.
  - `LogP` A class for calculating the LogP value of a molecule.
  - `TPSA` A class for calculating the TPSA value of a molecule.
  - `RotatableBonds` A class for calculating the number of the rotatable bonds of a molecule.
  - `HydrogenBondAcceptors` A class for calculating the number of the hydrogen bond acceptors of a molecule.
  - `HydrogenBondDonors` A class for calculating the number of the hydrogen bond donors of a molecule.
  - `Stereocenters` A class for calculating the number of stereocenters of a molecule.

- Following Pytests were created:
  - `test_mdonatello.py` for the testing of `mdonatello.py`
  - `test_drawer.py` for the testing of `drawer.py`
  - `test_mapper.py` for the testing of `mapper.py`

- Documentation was created an uploaded to:
    - [https://mdonatello.readthedocs.io/](url)

- Following Workflow files were created:
  - `Linux_CI_CD.yml ` for Linux based systems
  - `MacOS_CI_CD.yml ` for MacOS based systems
  - `Windows_CI_CD.yml ` for Windows based systems

talagayev committed:
Update test_mdonatello.py

Added Tests for Ipywidget initialization and Checkbox interaction update

talagayev committed:
Update mdonatello.py

Added MoleculeVisualizer

talagayev committed:
Update mdonatello.py

Added pharmacophore option, removed highlight aromatic rings and removed redundant calcuation of hydrogen bond donors and acceptors

talagayev committed:
Update mdonatello.py

Fix error due to get_hbond function being present, although it was removed

talagayev committed:
Update mdonatello.py

fixed error due to highlight_aromatic rings

talagayev committed:
Update mdonatello.py

Addition of Checkboxes for pharmacophores

talagayev committed:
Update __init__.py

Addition of importing of MoleculeVisualizert to init

richardjgowers commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+        
+    def display_molecular_weight(self, ag):
+        mw = Descriptors.MolWt(ag)
+        return HTML("<p style='margin: 0; margin-left: 100px;'>Molecular Weight: {:.2f} g/mol</p>".format(mw))
```
can you use f-string formatting on these HTML calls?

talagayev replied:
Yes can do, will adjust it

cbouysset replied:
since the molecule object is accessible as an attribute of the class I'd directly use `self.mol_noh` instead of passing it as a parameter for all these methods, otherwise it's best practice to decorate them with `@staticmethod` when methods don't need to access `self` attributes (but then do you really need them all in a class?). But I'd rather have them use `self.mol_noh` tbh so that you can have this in `update_display`:
```python
children.extend([
    self.display_num_h_donors(),
    self.display_num_h_acceptors(),
])
```

talagayev replied:
Sounds tricky , but I think I can try to adjust it, if not I would ask it on Friday.

richardjgowers commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+        rotatable_bonds = Descriptors.NumRotatableBonds(ag)
+        return HTML("<p style='margin: 0; margin-left: 100px;'>Number of Rotatable Bonds: {:.0f}</p>".format(rotatable_bonds))
+        
+    def save_selected_molecule(self, _):
```
probably filename as an argument here right?  In general if you want to use print, you probably want to instead do a logging call

richardjgowers commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
 import base64
 
 
+class MoleculeVisualizer:
```
add in a class docstring here that includes
- a description of what this class does
- example(s) of using this class

talagayev replied:
Ah yes, my bad, will do it :)

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
 
+class MoleculeVisualizer:
+    def __init__(self, ag, show_atom_indices=False, highlight_aromatic=False):
+        self.ag = ag.convert_to("RDKit")
```
nitpick: I find the `ag` and `ag_noh` confusing since they're not AtomGroup objects but rdkti molecules.
suggestion: Could you rename these as `mol` and `mol_noh` instead?

talagayev replied:
true, yes makes sense, that I can do

talagayev replied:
Yes, can do :)

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+        self.physiochem_props_checkbox.observe(self.update_display, names="value")
+        self.hbond_props_checkbox.observe(self.update_display, names="value")
+
+    def get_hbond_description(self):
```
Feel free to directly delete this since it's not really useful from the looks of it

talagayev replied:
The `get_hbond_description`? Yes it was twice, since it replicated the same things `display_num_h_acceptors` and `display_num_h_donors` do so I removed it in #3.

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+        num_h_acceptors = Descriptors.NumHAcceptors(self.ag_noh)
+        return f"Show H-Bond Donors/Acceptors"
+
+    def display_molecule(self, ag, show_atom_indices, highlight_aromatic):
```
Since this is already decoupled from the class itself, I think it'd make sense to move it outside as a separate function or class, so that you have one func/class for generating the depiction, and one class (the current one) that just wraps all the configuration around a widget. This will also make testing a bit easier.
Just for consistency if you go that route I'd return the raw string (i.e. SVG/PNG not wrapped with `HTML`) in that function, and then just have something like this in the class:
```python
def display_molecule(self):
    return HTML(
        draw_molecule(
            self.mol_noh,
            show_atom_indices=self.show_atom_indices_checkbox.value,
            highlight_aromatic=self.highlight_aromatic_checkbox.value,
        )
    )
```
And also make sure to use that `draw_molecule` function in `save_selected_molecule` as well

talagayev replied:
True, that makes sense and would be easier for testing, since then I could split up the testing more easier and the code would be more clear.

Hm, yes that looks more readable :)

For the `save_selected_molecule` I would also need to check Oliver's sugesstion on the Discord regarding the high quality figures to see if I can somehow implement it in the code here.

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+        html = f'<img src="data:image/png;base64,{base64.b64encode(bio.getvalue()).decode()}" />'
+        return HTML(html)
+        
+    def update_display(self, _=None):
```
I'd keep the same function signature as in the `ipywidgets` [docs examples](https://ipywidgets.readthedocs.io/en/7.6.2/examples/Widget%20Events.html#Signatures) for consistency, i.e. rename `_` to `change`, also I don't think you need a default value?

talagayev replied:
Yes the default value is not necessary here, so I can remove it

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+        self.show_atom_indices_checkbox = Checkbox(value=show_atom_indices, description="Show atom indices")
+        self.highlight_aromatic_checkbox = Checkbox(value=highlight_aromatic, description="Highlight aromatic atoms")
+        self.physiochem_props_checkbox = Checkbox(value=False, description="Show Physiochemical Properties")
+        self.hbond_props_checkbox = Checkbox(value=False, description=self.get_hbond_description())
```
Replace the description with the actual string directly

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+        return HTML(html)
+        
+    def update_display(self, _=None):
+        self.hbond_props_checkbox.description = self.get_hbond_description()
```
no need for this update

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+            for atom in ag.GetAtoms():
+                atom.ClearProp("atomNote")
+        
+        bio = BytesIO()
```
for PNG instead of using `Draw.MolToImage` and writing to a bytesIO object, you could directly use the `rdMolDraw2D.MolDraw2DCairo` class (that has the same API as the SVG one), this should simplify the code quite a bit

talagayev replied:
Yes that's true, will adjust it :)

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+        return f"Show H-Bond Donors/Acceptors"
+
+    def display_molecule(self, ag, show_atom_indices, highlight_aromatic):
+        if highlight_aromatic:
```
I don't think there's a reason to change which backend is used (png or svg) depending on whether the user wants to highlight aromatic atoms or show atom indices, e.g. you wouldn't be able to have both in the current case. See my comment below about using `rdMolDraw2D.MolDraw2DCairo` instead. For me SVG/PNG should be an option set by the user.

talagayev replied:
Should be doable, would it then be the case that if the user selects PNG it goes through `rdMolDraw2D.MolDraw2DCairo` and in the case the `rdMolDraw2D.MolDraw2DSVG` (would need to check if that makes sense what I wrote, but will try to do it)

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+                bond.GetIdx() for bond in ag.GetBonds() 
+                if bond.GetBeginAtom().GetIsAromatic() and bond.GetEndAtom().GetIsAromatic()
+            ]
+            d = rdMolDraw2D.MolDraw2DSVG(300, 300)
```
suggestion: should the image size be an option?

talagayev replied:
Yes, I can set them as default 300, 300 and put them as an optional input for the Visualization class, or do you mean the option for the user to adjust them via Ipywidgets, where they would be able to enter their numbers and this would adjust the figure accordingly?

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+                if bond.GetBeginAtom().GetIsAromatic() and bond.GetEndAtom().GetIsAromatic()
+            ]
+            d = rdMolDraw2D.MolDraw2DSVG(300, 300)
+            d.drawOptions().addStereoAnnotation = True
```
suggestion: should the draw options be accessible to the user somehow (maybe in another PR though to keep things simple here)?

talagayev replied:
Yes, that could be an option, can add it in a different PR as you suggested

cbouysset commented on mdonatello/mdonatello.py:
```diff
@@ -9,3 +9,138 @@
+            svg = d.GetDrawingText()
+            return HTML(svg)
+        
+        if show_atom_indices:
```
You could also just use `d.drawOptions().addAtomIndices = True`

talagayev replied:
Ah yes, that makes it easier :)

cbouysset commented:
Nice work Valerij! My main comments are basically splitting the drawing code from the widget class to make things easier to test, properly making use of class attributes in the different `display_*` methods instead of passing those attributes as parameters, and using the `rdMolDraw2D.MolDraw2D{SVG/Cairo}` class exclusively instead of relying on `MolToImage`.

cbouysset commented:
Also as some general feedback:
- for some of the comments that we will make on your PRs, feel free to open issues and fix them in later PRs if they are quite large in scope, just make sure to reference them somewhere.
- opening issues in general just to keep a track of what features/tests/docs need to be added, you can always label them as `wontdo` and close them if things change. This also gives some visibility to what is currently ongoing/planned for people interested in your project in general that may not be involved in our meetings, and is general "good practice"

talagayev commented:
> Also as some general feedback:
> 
> * for some of the comments that we will make on your PRs, feel free to open issues and fix them in later PRs if they are quite large in scope, just make sure to reference them somewhere.
> * opening issues in general just to keep a track of what features/tests/docs need to be added, you can always label them as `wontdo` and close them if things change. This also gives some visibility to what is currently ongoing/planned for people interested in your project in general that may not be involved in our meetings, and is general "good practice"

Yes makes sense, I would then open Issues and link them to this PR with the mention what I would adjust and then split up the big changes in separate Issues and the smaller cases, where it is easier to adjust in one Issue, where I mention them. I can also add labels to them, so that it is easier to see, that those are linked to the PR comments and which are the bigger ones and which are a combination of small fixes :)

talagayev committed:
Update mdonatello.py

changed ag and ag_noh to mol and mol_noh in case of rdkit mol objects

talagayev committed:
Update mdonatello.py

Change the atom indices to
d.drawOptions().addAtomIndices = show_atom_indices

talagayev committed:
Update mdonatello.py

Addition of width and height selection options

talagayev committed:
Update mdonatello.py

Added docstrings for MoleculeVisualizer as an example, would be modified later on

talagayev committed:
Update mdonatello.py

intend error + assignin default values

talagayev committed:
Update mdonatello.py

Fixed error with missing width and height input parameters

talagayev committed:
Update mdonatello.py

adjusted to use directly mol_noh

talagayev committed:
Update mdonatello.py

adjusted the separate display functions to use as input only self

talagayev committed:
Update mdonatello.py

changed the HTML return style to f string

talagayev committed:
Update mdonatello.py

Added logging call instead of print

talagayev committed:
Update mdonatello.py

Splitting up Drawer and displaying molecule into two functions

talagayev committed:
Update mdonatello.py

fixed error with missing bracket

talagayev committed:
Update mdonatello.py

fixed typo error with bracket

talagayev committed:
Update mdonatello.py

Added multiple ligand support + adjusted checkbox positions

talagayev committed:
Update mdonatello.py

Added HBox import

talagayev committed:
Update mdonatello.py

Adjusted Docstring style

talagayev committed:
Update mdonatello.py

Change to have the pharmacophore feature checkboxes present, instead of creating them dynamically

talagayev committed:
Update mdonatello.py

adjusted the pharmacophore feats

talagayev committed:
Update mdonatello.py

Added feats

talagayev committed:
Merge pull request #15 from talagayev/9_fixing_multiple_ligand_recognition

Adjustment of pharmacophore checkboxes

talagayev committed:
Merge pull request #2 from talagayev/MDonatello_tests

Test Addition to Ipywidget

talagayev committed:
Merge pull request #11 from talagayev/5_update_display_noh

Adjustment of update_display function and f string addition

talagayev committed:
Merge pull request #12 from talagayev/6_logging_call

Addition of a logging call to save_selected_molecule

talagayev committed:
Merge pull request #13 from talagayev/7_splitting_display_draw

Splitting the display_molecule into display and draw functions

talagayev committed:
Update mdonatello.py

Adjustment for the aromatic pharmacophore to use the isAromatic instead

talagayev committed:
Update mdonatello.py

First implementation of functional groups

talagayev committed:
Update mdonatello.py

Added additional functional groups and made color adjustments according to the element present

talagayev committed:
Update mdonatello.py

Added bonds coloring

yuxuanzhuang commented:
I believe @exs-cbouy mentioned the idea of segregating the display functionality from this widget class https://github.com/talagayev/MDonatello/pull/1#discussion_r1634395060

I'm considering whether it might be beneficial to more explicitly define a `DisplayedMolecule` class along with a `Property` class. This could potentially be overkill, but it probably would provide clearer structure. Additionally, it would offer more flexibility for incorporating additional display options and doing caching to avoid query properties multiple times.

```python
from functools import cached_property

class Property:
    name = 'property'
    def __init__(self, mol):
        self.mol = mol

    # you only need to query this property once.
    @cached_property
    def property_value(self):
        raise NotImplementedError('')

    def __repr__(self):
        return f"<p style='margin: 0; margin-left: 100px;'>Number of {self.name}: {self.property_value}</p>"

class RotatableBonds(Property):
    name = 'Rotatable Bonds'

    @cached_property
    def property_value(self):
        return Descriptors.NumRotatableBonds(self.mol)
```

yuxuanzhuang commented:
Another point to consider relates to the current behavior of the `display` method, which only activates when an instance is created. To enable re-displaying of the widget, you might define the `__repr__` method as follows:
```python
def __repr__(self):
    display(self.output_dropdown, self.output)
    return ''
```
This ensures that whenever the representation of the object is requested, the widgets are displayed.

I also have a look at `nglview`, it inherits the `DOMWidget` class as seen in its [source code](https://github.com/nglviewer/nglview/blob/master/nglview/widget.py#L13). For more details on custom widgets, you can refer to [this guide](https://minrk-ipywidgets.readthedocs.io/en/latest/examples/Widget%20Custom.html).

Not sure if there's an advantage with its implementation.

talagayev committed:
Update mdonatello.py

Added Black formatting

talagayev committed:
Update mdonatello.py

starting doing Type Checking&Hinting

talagayev committed:
Update mdonatello.py

Type Checking&Hinting

talagayev committed:
Update mdonatello.py

restructure of code for physiochem properties and hbond acceptors donors

talagayev committed:
Update mdonatello.py

moved pharmacophore and functional groups to separate classes

talagayev committed:
Update mdonatello.py

fixed functional group error

talagayev committed:
Update mdonatello.py

talagayev committed:
Update mdonatello.py

moved highlighting of molecules

talagayev committed:
Update mdonatello.py

moved highlight and drawer

talagayev committed:
Update mdonatello.py

fixed error and changed func group representation

talagayev committed:
Update mdonatello.py

changed functional groups to checkboxes again

talagayev committed:
Update mdonatello.py

fixed error for input

talagayev committed:
Update mdonatello.py

adjusted the functional group highlighting of the bonds

talagayev committed:
Update mdonatello.py

Added rotatable bonds

talagayev committed:
Update mdonatello.py

Adjusted the printing of the properties to have correct values

talagayev committed:
Update test_mdonatello.py

Added tests for property classes

talagayev committed:
Update __init__.py

talagayev committed:
Update mdonatello.py

Added partial charges and adjusted the code with black formatting

talagayev committed:
Update mdonatello.py

Added Murcko and Stereocenters and fixed Rotatable bonds recognition of atom indices

talagayev committed:
Moved functional group and pharmacophore classes to mapper.py

talagayev committed:
Separated the highlighting into classes and moved with MoleculeDrawer to drawer.py

talagayev committed:
Create properties.py

talagayev committed:
Adjusted mdonatello.py with the new code adjustment

talagayev committed:
Update mdonatello.py

talagayev committed:
Separated Moleculevisualizer in more classes, added documentation and removed not used imports

talagayev committed:
Create test_mapper.py

talagayev committed:
Create CI_CD.yml

talagayev committed:
Update CI_CD.yml

removed env update

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

Added codecov

talagayev committed:
Delete mdonatello/tests/test_mdonatello.py

currently removed to adjust new tests and test codecov

talagayev committed:
Update CI_CD.yml

adjusting code cov

talagayev committed:
Update CI_CD.yml

talagayev committed:
Create test_drawer.py

Adde first test in drawer

talagayev committed:
Update CI_CD.yml

talagayev committed:
Create test_mdonatello.py

talagayev committed:
Added test pdb file

talagayev committed:
Update test_mdonatello.py

talagayev committed:
Update test_mdonatello.py

talagayev committed:
Added new tests

talagayev committed:
Added Figure for Display.rst

talagayev committed:
Added Display in index.rst

talagayev committed:
Added display.rst

talagayev committed:
Update getting_started.rst

talagayev committed:
Update getting_started.rst

talagayev committed:
Added IpyWidgets Matrix

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Update CI_CD.yml

talagayev committed:
Add files via upload

talagayev committed:
Update conf.py

talagayev committed:
Delete docs/source/_static/logo/Mdonatello_logo.png

talagayev committed:
Add files via upload

talagayev committed:
Update conf.py

talagayev committed:
Delete docs/source/_static/logo/Mdonatello_logo.png

talagayev committed:
Add files via upload

talagayev committed:
Delete docs/source/_static/logo/Mdonatello_logo.png

talagayev committed:
Add files via upload

talagayev committed:
Update getting_started.rst

talagayev committed:
Update getting_started.rst

talagayev committed:
Update README.md

talagayev committed:
Update README.md

talagayev committed:
Update getting_started.rst

talagayev committed:
Create CI_CD_MacOS.yml

talagayev committed:
Rename CI_CD.yml to Linux_CI_CD.yml

talagayev committed:
Rename Linux_CI_CD.yml to CI_CD.yml

talagayev committed:
Update and rename CI_CD.yml to Linux_CI_CD.yml

talagayev committed:
Update Linux_CI_CD.yml

talagayev committed:
Update and rename CI_CD_MacOS.yml to MacOS_CI_CD.yml

talagayev committed:
Create Windows_CI_CD.yml

talagayev committed:
Update Windows_CI_CD.yml

talagayev committed:
Update Windows_CI_CD.yml

talagayev committed:
Merge pull request #19 from talagayev/Class_code_movement

Class code movement

talagayev committed:
Merge pull request #18 from talagayev/Rotatable_bond_addition

Rotatable bond addition

talagayev committed:
Merge pull request #17 from talagayev/Code_restructure

Code class restructure

talagayev committed:
Merge pull request #16 from talagayev/Functional_group_addition

Functional group addition

talagayev committed:
Merge pull request #14 from talagayev/8_multiple_ligand_recognition

Main PR: Multiple Ligand recognition

talagayev committed:
Merge pull request #20 from talagayev/7_splitting_display_draw

Combining all PR request into the Main_code_PR

talagayev committed:
Added Logging import

talagayev committed:
Added Documentation to the functions in drawer.py

talagayev committed:
Update drawer.py

talagayev committed:
indent error fix

talagayev committed:
Added documentation to mapper.py

talagayev committed:
Added documentation of functions to properties.py

talagayev committed:
Update README.md

talagayev committed:
Update README.md

talagayev committed:
Delete mdonatello/analysis directory due to not using it

talagayev committed:
Delete mdonatello/tests/analysis directory

talagayev committed:
Rename mdonatello/tests/test_files/input.pdb to mdonatello/data/input.pdb

Moved input.pdb to data

talagayev committed:
Changed path to find input.pdb

Changed path to

talagayev committed:
Create drawer.rst

talagayev committed:
Add files via upload

talagayev committed:
Delete docs/modules directory

talagayev committed:
Create drawer.rst

talagayev committed:
Add files via upload

talagayev committed:
Update api.rst

talagayev committed:
Update conf.py

talagayev committed:
Update README.md

Added mention of GSOC and Release Link

talagayev committed:
Update README.md
